contrib/net/http/trace.go

```diff
@@ -94,3 +95,15 @@ func (w *responseWriter) WriteHeader(status int) {
 	w.ResponseWriter.WriteHeader(status)
 	w.status = status
 }
+
+// HeaderTagsFromRequest matches req headers to user-defined list of header tags
+// and creates span tags based on the header tag target and the req header value
+func HeaderTagsFromRequest(req *http.Request, headersAsTags map[string]string) ddtrace.StartSpanOption {
+	return func(cfg *ddtrace.StartSpanConfig) {
+		for k := range req.Header {
+			if ht, ok := headersAsTags[strings.ToLower(k)]; ok && !strings.HasPrefix(strings.ToLower(k), "x-datadog-") {
+				cfg.Tags[ht] = strings.Join(req.Header.Values(k), ",")
+			}
+		}
```

Comment:
```suggestion
		for h, v := range req.Header {
			h = strings.ToLower(h)
			if tag, ok := headersAsTags[h]; ok && !strings.HasPrefix(h, "x-datadog-") {
				cfg.Tags[tag] = strings.Join(v, ",")
			}
		}
```
Would this work instead? No need to call out `req.Header.Values(k)` if you already have them as the second argument of the range (which is currently just being thrown away)